scraper.js

```diff
@@ -1,25 +1,27 @@
 const api = require('./api/api.js')
-const globals = require('./globals.js')
+const helpers = require('./helpers.js')
+const parser = require('./parser.js')
 
 const scraper = async (blockHeight) => {
   let blockHash = await api.getBlockHashByHeight(blockHeight)
-  let block = await api.getBlock(blockHash)
-  let transactions = JSON.parse(block).tx
+  let rawBlock = await api.getBlockByHash(blockHash)
+  let block = parser.blockParser(rawBlock)
   let blockTransactionData = []
 
   // skip the generation transaction (coinbase) when scraping
-  for (let i = 1; i < transactions.length; i++) {
-    let tx = await api.getRawTransaction(transactions[i])
-    tx = JSON.parse(tx)
+  for (let i = 1; i < block.transactions.length; i++) {
+    let rawTx = await api.getRawTransaction(block.transactions[i])
+    let tx = parser.txParser(rawTx)
 
-    let txAmount = globals.getTransactionTotal(tx.vout)
-    let fee = await globals.calculateFee(tx, txAmount)
-    let txTime = new Date(tx.time * 1000)
+    tx.total = tx.total || helpers.getTransactionTotal(tx.outputs)
```

Comment:
I like this direction... attributes which a user wants should always be populated, where possible - so checking if it is defined and then doing the calculation makes sense.

creating some more in depth decision trees about what fields to populate and parse based on user input would be next step